internal/cmd/hana/provision.go

```diff
@@ -0,0 +1,108 @@
+package hana
+
+import (
+	"context"
+	"fmt"
+	"github.com/kyma-project/cli.v3/internal/kube"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type hanaProvisionConfig struct {
+	ctx        context.Context
+	kubeClient kube.Client
+
+	kubeconfig  string
+	name        string
+	namespace   string
+	planName    string
+	memory      int
+	cpu         int
+	whitelistIP []string
+}
+
+func NewHanaProvisionCMD() *cobra.Command {
+	config := hanaProvisionConfig{}
+
+	cmd := &cobra.Command{
+		Use:   "provision",
+		Short: "Provisions a Hana instance on the Kyma.",
+		Long: `Use this command to provision a Hana instance on the SAP Kyma platform.
+`,
+		PreRunE: func(_ *cobra.Command, args []string) error {
+			return config.complete()
+		},
+		RunE: func(_ *cobra.Command, _ []string) error {
+			return runProvision(&config)
+		},
+	}
+
+	cmd.Flags().StringVar(&config.kubeconfig, "kubeconfig", "", "Path to the Kyma kubecongig file.")
+
+	cmd.Flags().StringVar(&config.name, "name", "", "Name of Hana instance.")
+	cmd.Flags().StringVar(&config.namespace, "namespace", "default", "Name of namespace.")
+	cmd.Flags().StringVar(&config.planName, "plan", "hana", "Name of the service plan.")
+	cmd.Flags().IntVar(&config.memory, "memory", 30, "??? memory")                                          //TODO: fulfill proper usage
+	cmd.Flags().IntVar(&config.cpu, "cpu", 2, "??? cpu")                                                    //TODO: fulfill proper usage
+	cmd.Flags().StringSliceVar(&config.whitelistIP, "whitelist-ip", []string{"0.0.0.0/0"}, "??? whitelist") //TODO: fulfill proper usage
```

Comment:
Can we add descriptions for these flags?

Reply:
added